### PR TITLE
funders:update to ROR v2

### PR DESCRIPTION
**Pull request by tmorrell**
:heart: Thank you for your contribution!

### Description

The PR switches the funders vocabulary to use the v2 version of the ROR schema https://github.com/inveniosoftware/invenio-vocabularies/issues/316.

It adds new fields:

- aliases (aliases listed in the ROR names field)
- status (status of record in ROR)
- types (type of organization)
- country_name (full name of first country)
- location_name (name of first location)
- acronym (first acronym listed in the ROR names field)

For compatibility purposes we only include the first location and acronym. There doesn't seem to be many RORs with multiple locations or acronyms. Ideally we'd be able to include all of these, but it would require a breaking change which doesn't seem to be worth it.

It no longer assumes that the primary name is in english (which isn't correct). This will result in many more `title` fields that don't have english values, since ROR doesn't currently have complete labeling of the languages of names. 

I'll moves the transformer to a central location so it can also be used for affiliations and handle `withdrawn` records in future PRs.

### Checklist

Ticks in all boxes and 🟢 on all GitHub actions status checks are required to merge:

- [x] I'm aware of the [code of conduct](https://inveniordm.docs.cern.ch/contribute/code-of-conduct/).
- [x] I've created [logical separate commits](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commits) and followed the [commit message format](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commit-message).
- [x] I've added relevant test cases.
- [ ] I've added relevant documentation.
- [ ] I've marked [translation strings](https://inveniordm.docs.cern.ch/develop/howtos/i18n/).
- [x] I've identified the [copyright holder(s)](https://inveniordm.docs.cern.ch/contribute/copyright-policy/) and updated copyright headers for touched files (>15 lines contributions).
- [x] I've NOT included third-party code (copy/pasted source code or new dependencies).
    * If you have added [third-party code (copy/pasted or new dependencies)](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#third-party-codedependencies), please reach out to an [architect](https://github.com/orgs/inveniosoftware/teams/architects/members).

**Frontend**

- [ ] I've followed the [CSS/JS](https://inveniordm.docs.cern.ch/develop/best-practices/css-js/) and [React](https://inveniordm.docs.cern.ch/develop/best-practices/react/) guidelines.
- [ ] I've followed the [web accessibility](https://inveniordm.docs.cern.ch/develop/best-practices/accessibility/) guidelines.
- [ ] I've followed the [user interface](https://inveniordm.docs.cern.ch/develop/best-practices/ui/) guidelines.

**Reminder**

By using GitHub, you have already agreed to the [GitHub’s Terms of Service](https://help.github.com/articles/github-terms-of-service/#6-contributions-under-repository-license) including that:

1. You license your contribution under the same terms as the current repository’s license.
2. You agree that you have the right to license your contribution under the current repository’s license.

### CODE REVIEW

File: invenio_vocabularies/contrib/funders/datastreams.py [hunk 1]
```diff
@@ -38,36 +39,62 @@ def apply(self, stream_entry, **kwargs):
...
+                funder["name"] = name["value"]
+            if "label" in name["types"]:
+                funder["title"][lang] = name["value"]
+            if "alias" in name["types"]:
+                aliases.append(name["value"])
+            if "acronym" in name["types"]:
+                # We only save the first acronyn, to maintain
+                # compatability with existing data structure
+                if not acronym:
+                    acronym = name["value"]
```
Comment (slint) [inline invenio_vocabularies/contrib/funders/datastreams.py:62]: minor: could we maybe add the extra acronyms to `aliases`?

File: invenio_vocabularies/contrib/funders/mappings/os-v1/funders/funder-v1.0.0.json [hunk 2]
```diff
@@ -57,6 +57,24 @@
...
+      },
+      "acronym": {
+        "type": "text"
+      },
+      "status": {
+        "type": "keyword"
+      },
+      "aliases": {
+        "type": "text"
+      },
```
Comment (slint) [inline invenio_vocabularies/contrib/funders/mappings/os-v1/funders/funder-v1.0.0.json:74]: comment: maybe mot for now, but we'll have to test how search behaves (probably already much better than before :) ), and possibly consider also adding a `acronym.keyword` subfield to allow for exact search matches.

**Review by slint — APPROVED** [review]
LGTM! I think we can merge, just an optional suggestion to abuse the `aliases` field to store more acronyms.

File: invenio_vocabularies/contrib/funders/datastreams.py (hunk 1 above)
Comment (tmorrell) [reply invenio_vocabularies/contrib/funders/datastreams.py]: Ah, that's a good idea. Added!

File: invenio_vocabularies/contrib/funders/mappings/os-v1/funders/funder-v1.0.0.json (hunk 2 above)
Comment (tmorrell) [reply invenio_vocabularies/contrib/funders/mappings/os-v1/funders/funder-v1.0.0.json:74]: I suspect there is going to be an additional task on improving search performance. `affiliations` uses `text`, so I kept it the same here. Can I include multiple mappings?

Comment (slint) [reply invenio_vocabularies/contrib/funders/mappings/os-v1/funders/funder-v1.0.0.json:74]: Agree, we should have a look at search performance across all vocabularies and tune them all at the same time. Besides adding some types for allowing for more precise queries, we should also e.g. have a look at what fields are used for free-text search, using the [`settings.index.query.default_field`](https://github.com/inveniosoftware/invenio-rdm-records/blob/7f465e49eaec29a75e14d0b38ebc2117beabfe23/invenio_rdm_records/records/mappings/os-v2/rdmrecords/records/record-v6.0.0.json#L1425-L1453)

**Pull request merged**